test/unit/plugins/rest_spec.js

```diff
@@ -131,40 +147,180 @@ describe("The Rest plugin", function () {
+				AuthError.name = "AuthFailure";
+
+				runInstancesStub = Sinon.stub(awsAdapter, "runInstances")
+				.rejects(AuthError);
+			});
+
+			after(function () {
+				runInstancesStub.restore();
+			});
+
+			it("returns an error with statusCode 400", function () {
+				var request = new Request("POST", "/v1/instances").mime("application/json").payload({
+					ami  : VALID_AMI,
+					type : VALID_TYPE
 				});
+				return request.inject(server)
+				.then(function (response) {
+					expect(response.statusCode, "status").to.equal(500);
+					expect(response.payload)
+					.to.equal("Authentication Failure. Ensure your AWS credentials have been correctly used.");
+				});
+			});
+		});
+
+		describe("with an ami that doesn't exist", function () {
+
+			before(function () {
+				var AMIError  = new Error();
+				AMIError.name = "InvalidAMIID.Malformed";
+
+				runInstancesStub = Sinon.stub(awsAdapter, "runInstances")
+				.rejects(AMIError);
+			});
 
-				return new Request("POST", "/v1/instances").inject(server)
+			after(function () {
+				runInstancesStub.restore();
+			});
+
+			it("returns an error with statusCode 400", function () {
+				var request = new Request("POST", "/v1/instances").mime("application/json").payload({
+					ami  : "ami-invalid",
+					type : VALID_TYPE
+				});
+				return request.inject(server)
 				.then(function (response) {
-					result = response;
+					expect(response.statusCode, "status").to.equal(400);
+					expect(response.payload)
+					.to.equal("The AMI entered does not exist. Ensure it is of the form ami-xxxxxx.");
+				});
+			});
+		});
+
+		describe("with a type that doesn't exist", function () {
+
+			before(function () {
+				var TypeError  = new Error();
+				TypeError.name = "InvalidParameterValue";
+
+				runInstancesStub = Sinon.stub(awsAdapter, "runInstances")
+				.rejects(TypeError);
+			});
+
+			after(function () {
+				runInstancesStub.restore();
+			});
+
+			it("returns an error with statusCode 400", function () {
+				var request = new Request("POST", "/v1/instances").mime("application/json").payload({
+					ami  : VALID_AMI,
+					type : "t2.notExist"
+				});
+				return request.inject(server)
+				.then(function (response) {
+					expect(response.statusCode, "status").to.equal(400);
+					expect(response.payload)
+					.to.equal("The Type entered does not exist. Ensure it is a valid EC2 type.");
+				});
+			});
+		});
+
+		describe("when there is a problem with the database connection", function () {
+			var mapper = new MemoryMapper();
+			var server = new Hapi.Server();
+
+			before(function () {
+				Sinon.stub(mapper, "create").rejects(new Error("Simulated Failure."));
+				server.connection();
+				return server.registerAsync({
+					register : Rest,
+					options  : {
+						mapper : mapper
+					}
 				});
 			});
 
 			after(function () {
 				mapper.create.restore();
+				return server.stopAsync();
 			});
 
 			it("displays an error page", function () {
```

Comment:
Is this description still accurate? This looks like it is for a REST API and not a web UI.